modules/forum/src/main/TopicRepo.scala

```diff
@@ -60,6 +61,15 @@ final class TopicRepo(val coll: Coll, filter: Filter = Safe)(implicit
     }
   }
 
+  def findDuplicate(topic: Topic): Fu[Option[Topic]] =
+    coll.one[Topic](
+      $doc(
+        "createdAt" $gt DateTime.now.minusHours(1),
+        "userId" -> ~topic.userId,
+        "name"   -> topic.name
+      )
+    )
+
```

Comment:
there are no indexes for that in DB, so it will be very slow in prod.

Instead of hitting the DB, you could use an [ExpireSetMemo](https://github.com/ornicar/lila/blob/master/modules/memo/src/main/ExpireSetMemo.scala) to memorize the `$name:${topic.name}` created for a short duration.

Usage example: https://github.com/ornicar/lila/blob/ae29bb9b5147065f8d8537b5983c3872ebadb3af/modules/puzzle/src/main/PuzzleApi.scala#L172-L181